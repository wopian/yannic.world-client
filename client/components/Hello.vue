<template lang='pug'>
  .hello
    h1 {{ msg }}
</template>

<script>
export default {
  name: 'hello',
  data () {
    return {
      msg: 'Welcome to Your Vue.js App'
    }
  }
}
</script>
