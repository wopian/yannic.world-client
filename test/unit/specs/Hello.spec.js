import Vue from 'vue'
import Hello from 'client/components/Hello'

const Constructor = Vue.extend(Hello)
const vm = new Constructor().$mount()

describe('Hello.vue', () => {
  it('should render correct contents', () => {
    expect(vm.$el.querySelector('.hello h1').textContent)
      .toBe('Welcome to Your Vue.js App')
  })

  it('renders to a nice snapshot', () => {
    const renderer = require('vue-server-renderer').createRenderer()
    renderer.renderToString(vm, (err, str) => {
      if (err) console.error(err)
      expect(str).toMatchSnapshot()
    })
  })
})
