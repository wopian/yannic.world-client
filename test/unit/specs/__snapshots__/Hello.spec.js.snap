// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hello.vue renders to a nice snapshot 1`] = `
<div server-rendered="true" class="hello">
    <h1>Welcome to Your Vue.js App</h1>
</div>
`;
